src/services/azureFunctionsService.ts

```diff
@@ -68,16 +69,34 @@ export class AzureFunctionsService implements mssql.IAzureFunctionsService {
 			return;
 		}
 		let projectFile = await azureFunctionUtils.getAzureFunctionProject();
+		let newHostProjectFile: azureFunctionUtils.IFileFunctionObject;
+		let hostFile: string;
+
 		if (!projectFile) {
 			let projectCreate = await vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsProjectMustBeOpened,
 				LocalizedConstants.createProject, LocalizedConstants.learnMore);
 			if (projectCreate === LocalizedConstants.learnMore) {
 				vscode.commands.executeCommand('vscode.open', vscode.Uri.parse(constants.sqlBindingsDoc));
+				return;
 			} else if (projectCreate === LocalizedConstants.createProject) {
-				// start the create azure function project flow
-				await azureFunctionApi.createFunction({});
+				try {
+					// start the create azure function project flow
+					// create a watcher to see if the host file is created after the flow is complete
+					newHostProjectFile = await azureFunctionUtils.waitForNewHostFile();
+					await azureFunctionApi.createFunction({});
+					const timeoutForHostFile = timeoutPromise(LocalizedConstants.timeoutProjectError);
+					hostFile = await Promise.race([newHostProjectFile.filePromise, timeoutForHostFile]);
+					if (hostFile) {
+						// start the add sql binding flow
+						projectFile = await azureFunctionUtils.getAzureFunctionProject();
+					}
+				} catch (error) {
+					vscode.window.showErrorMessage(utils.formatString(LocalizedConstants.errorNewAzureFunction, error.message) ?? error);
```

Comment:
I don't see the format index in the string - pretty sure that's required for this to work correctly.

Reply:
I used it the same way we use it in utils file. https://github.com/microsoft/vscode-mssql/blob/main/src/models/utils.ts#L186

Reply:
What do you mean "used it in the utils file". It doesn't get used there. It's defined there...

Look at the logic - it's going through and searching for instances of `{\d+}` (e..g {0}, {1}, etc.) and replacing them with the values you pass in.

Just find other uses of utils.formatString and look at how those strings are defined.

Reply:
This is how it is defined. You pass in multiple string to format together. This is also in azurefunctionutils in the same way.

Reply:
https://github.com/microsoft/vscode-mssql/blob/d628907b30b49760edc87fd90276dd121af27572/src/azureFunction/azureFunctionUtils.ts#L46

Reply:
I'm saying that the string itself is wrong - it needs to have the index specifier `{0}` for the replacement to work.

e.g. for failedToParse https://github.com/microsoft/vscode-mssql/blob/main/localization/xliff/enu/constants/localizedConstants.enu.xlf#L531 it looks like this

`Failed to parse "{0}": {1}.`

You currently have the string defined as just 

`Error creating new Azure Function:`

in [localizedConstants.enu.xlf](https://github.com/microsoft/vscode-mssql/blob/vabhog/newProj/localization/xliff/enu/constants/localizedConstants.enu.xlf#L579). which means the function will never do the replacement to add the error text. It needs to be 

`Error creating new Azure Function: {0}`

Reply:
Ah, I did miss that. Thanks for clarifying, I will be sure to make a note of this when addressing future work. Updated the localizedConstant.